whipper/common/mbngs.py

```diff
@@ -161,10 +161,82 @@ def _getWorks(recording):
     if 'work-relation-list' in recording:
         for work in recording['work-relation-list']:
             if work['type-id'] in valid_work_rel_types:
-                works.append(work['work']['id'])
+                works.append(work['work'])
     return works
 
 
+def _getWorksId(works):
+    """Get work ids from "performance of" works"""
+    ids = set()  # using a set to prevent duplicates
+    for work in works:
+        ids.add(work['id'])
+    # convert back to list since mutagen doesn't support set
+    return list(ids)
+
+
+def _getComposers(works):
+    """
+    Extract composer(s) from works' artist-relation-list into a list
+
+    :rtype: list
+    """
+    composers = set()  # using a set to prevent duplicates
+    valid_artist_rel_type = [
+        'd59d99ea-23d4-4a80-b066-edca32ee158f',  # "Composer"
+    ]
+    for work in works:
+        if 'artist-relation-list' in work:
+            for artist_relation in work['artist-relation-list']:
+                if artist_relation['type-id'] in valid_artist_rel_type:
+                    composerName = artist_relation['artist']['name']
+                    composers.add(composerName)
+    # convert back to list since mutagen doesn't support set
+    return list(composers)
+
+
+def _getPerformers(recording):
+    """
+    Extract performer(s) from recordings' artist-relation-list
+
+    :rtype: list
+    """
+    performers = []
+    valid_type_id = [
+        '59054b12-01ac-43ee-a618-285fd397e461',  # "Instruments"
+        '0fdbe3c6-7700-4a31-ae54-b53f06ae1cfa',  # "Vocals"
+        '628a9658-f54c-4142-b0c0-95f031b544da'   # "Performers"
+    ]
+    common_attribute_name = [
+        'guest', 'additional', 'solo'
+    ]
+    if 'artist-relation-list' in recording:
+        for artist_relation in recording['artist-relation-list']:
+            if artist_relation['type-id'] in valid_type_id:
+                type = artist_relation['type']
+                name = artist_relation['artist']['name']
+
+                if 'attribute-list' in artist_relation:
```

Comment:
This whole `attribute-list` parsing is still subject to discussion about feature bloat and purpose of whipper. My take is that we probably want this to be super simple if we want it (so just a list of names, no instruments etc.), and @JoeLametta so far agreed with this. Still no comment from @MerlijnWajer on this question. This is the only thing I would probably want changed about this PR at this point.

Reply:
Done! :wink: